.circleci/config.yml

```diff
@@ -24,29 +28,24 @@ jobs:
           name: Create required directories
           command: |
             mkdir -p $TEST_REPORTS_DIR/ ~/.virtualenvs ~/.ssh
-            # One needs to concateante here any files that can alter the virtualenv in a way:
+            # One needs to concatenate here any files that can alter the virtualenv in a way:
             #   - setup.py: for the endpoints installed in $VIRTUAL_ENV/bin
             #   - requirements.txt
             #   - dev-requirements.txt
             # Since we do a develop install of the code itself (cf. the '-e .' in dev-requirments.txt)
             # it does not matter.
+            pyenv local 3.5.2
             cat tests/requirements.txt tests/requirements.yml > circleci-requirements.txt
+            # In case Python's version in the image changes. The pyenv command above should
+            # break but we still need to regenerate the cache afterwards)
+            echo $(python3 --version) >> circleci-requirements.txt
             # Note that if the workflow is split, you will need to have the file `venv/circleci-requirements.txt`
             # passed from one build job to the other, either through caching or using workspaces.
+            sudo apt-get update
+            sudo apt-get install "nginx-light=${NGINX_VERSION}"
```

Comment:
Makes tests fairly slow. Maybe doing as with TravisCI would be faster (VM -> Docker -> Tests). Though seems doubtful as we do that in the docker container too. Would need to build a custom testing image.